tests/test_kv_store.c

```diff
@@ -60,11 +102,77 @@ static void test_type_H(void **state)
 	}
 }
 
+static void test_kvstore_iterate(void **state)
+{
+	struct iovec           test_iov[_KV_VALUE_IDX_COUNT];
+	size_t                 data_size, kv_size;
+	struct iovec *         return_iov;
+	kv_store_iter_t *      iter;
+	kv_store_value_flags_t flags        = KV_STORE_VALUE_REF | KV_STORE_VALUE_VECTOR;
+	uint64_t               seqnum       = 0;
+	sid_resource_t *       kv_store_res = NULL;
+	struct kv_update_arg   update_arg;
+	size_t                 meta_size = 0;
+
+	kv_store_res = sid_resource_create(SID_RESOURCE_NO_PARENT,
+	                                   &sid_resource_type_kv_store,
+	                                   SID_RESOURCE_RESTRICT_WALK_UP,
+	                                   "testkvstore",
+	                                   &main_kv_store_res_params,
+	                                   SID_RESOURCE_PRIO_NORMAL,
+	                                   SID_RESOURCE_NO_SERVICE_LINKS);
+
+	KV_VALUE_PREPARE_HEADER(test_iov, seqnum, flags, (char *) TEST_OWNER);
```

Comment:
Here, for the `KV_VALUE_PREPARE_HEADER`, the flags passed are supposed to be of `sid_ucmd_kv_flags_t` type - these are the flags that define permissions and persistency (protected/private/reserved...) - this is the extra part that ubridge adds on top of the pure `kv_store_value`.

So it's the `struct kv_store_value` that `kv_store.c` defines:

  [size] [int_flags] [ext_flags] [kv_store_data]

Then the `data` from the `kv_store_value` above is further enriched in `ubridge.c` as `struct kv_value`:

  [seqnum] [flags] [data]

So the overall record is then actually:
  

  [size] [int_flags] [ext_flags] [kv_store_data == [seqnum] [flags] [data]]